packages/uiweb/src/lib/context/spacesContext.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { createContext, useContext, useState } from 'react';
+
+interface SpaceDataContext {
+  trendingListData: any;
+  setTrendingListData: React.Dispatch<React.SetStateAction<any>>;
+  spaceBannerData: any;
+  setSpaceBannerData: React.Dispatch<React.SetStateAction<any>>;
+}
+
+const SpaceDataContext = createContext<SpaceDataContext | undefined>(undefined);
+
+export const useSpaceData = (): SpaceDataContext => {
+  const context = useContext(SpaceDataContext);
+  if (!context) {
+    throw new Error('useSpaceData must be used within a SpaceDataProvider');
+  }
+  return context;
+}
```

Comment:
lets move this custom hook to a separate hooks folder, as there will be future addition of custom hooks as well as we progress in this project.